src/util/meson.build

```diff
@@ -23,6 +23,16 @@ if is_developer_build
     util_src += files('debug.c')
 endif
 
+use_designated_init = cc.compiles('''
+struct { ; } __attribute__((designated_init)) x;
```

Comment:
Empty structs are a GNU extension, you should add a dummy member.

Reply:
Fixed in a1b4729.